client/my-sites/plans-features-main/hooks/use-is-custom-domain-allowed-on-free-plan.ts

```diff
@@ -1,12 +1,23 @@
-import config from '@automattic/calypso-config';
 import { getTld } from 'calypso/lib/domains';
+import { useExperiment } from 'calypso/lib/explat';
+import type { DataResponse } from 'calypso/my-sites/plan-features-2023-grid/types';
 
-const useIsCustomDomainAllowedOnFreePlan = ( domainName?: string ) => {
-	if ( ! domainName ) {
-		return false;
-	}
+const useIsCustomDomainAllowedOnFreePlan = (
+	flowName?: string | null,
+	domainName?: string
+): DataResponse< boolean > => {
+	const [ isLoadingAssignment, experimentAssignment ] = useExperiment(
+		'calypso_onboarding_plans_dotblog_on_free_plan_202307',
+		{
+			isEligible:
+				flowName === 'onboarding' && domainName != null && getTld( domainName ) === 'blog',
+		}
+	);
 
-	return config.isEnabled( 'domains/blog-domain-free-plan' ) && getTld( domainName ) === 'blog';
+	return {
+		isLoading: isLoadingAssignment,
+		entry: experimentAssignment?.variationName === 'treatment',
```

Comment:
Can you explain the meaning behind the word `entry`?

Reply:
It's just inherited from the previous `SingleFreeDomainSuggestion` type. `result` sounds better so I've updated it by https://github.com/Automattic/wp-calypso/pull/79279/commits/4304cbd2d651f4c8d82513653a0ab1a83ed6c0b6 :)